app/models/repository_host/github.rb

```diff
@@ -231,56 +231,42 @@ def gather_maintenance_stats
         return []
       end
 
-      client = AuthToken.v4_client
+      exists = !Github.fetch_repo(repository.full_name).nil?
+      repository.download_issues
+      repository.download_pull_requests
+
+      # use api_client methods?
+      v4_client = AuthToken.v4_client
       v3_client = AuthToken.client({auto_paginate: false})
       now = DateTime.current
 
       metrics = []
 
-      result = MaintenanceStats::Queries::FullRepoQuery.new(client).query( params: {owner: repository.owner_name, repo_name: repository.project_name} )
-      unless check_for_v4_error_response(result)
-        metrics << MaintenanceStats::Stats::Github::IssueRates.new(result).get_stats
-        metrics << MaintenanceStats::Stats::Github::PullRequestRates.new(result).get_stats
-        metrics << MaintenanceStats::Stats::Github::AverageCommitDate.new(result).get_stats
-      end
-
-      result = MaintenanceStats::Queries::RepoReleasesQuery.new(client).query( params: {owner: repository.owner_name, repo_name: repository.project_name, end_date: now - 1.year} )
+      result = MaintenanceStats::Queries::Github::RepoReleasesQuery.new(v4_client).query( params: {owner: repository.owner_name, repo_name: repository.project_name, end_date: now - 1.year} )
       unless check_for_v4_error_response(result)
         metrics << MaintenanceStats::Stats::Github::ReleaseStats.new(result).get_stats
       end
 
-      result = MaintenanceStats::Queries::CommitCountQuery.new(client).query(params: {owner: repository.owner_name, repo_name: repository.project_name, start_date: (now - 1.week).iso8601} )
+      result = MaintenanceStats::Queries::Github::CommitCountQuery.new(v4_client).query(params: {owner: repository.owner_name, repo_name: repository.project_name, start_date: (now - 1.week).iso8601} )
```

Comment:
I would also like to get this down to one GraphQL query, but have not had a chance to try and figure it out yet.